spatialmath/points.go

```diff
@@ -0,0 +1,299 @@
+package spatialmath
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+
+	"github.com/golang/geo/r3"
+	commonpb "go.viam.com/api/common/v1"
+)
+
+// points is a collision geometry that represents a collection of 3D points, it has a pose and points that fully define it.
+type points struct {
+	pose   Pose
+	label  string
+	points []r3.Vector
+}
+
+// NewPoints instantiates a new points Geometry.
+func NewPoints(pose Pose, pts []r3.Vector, label string) (Geometry, error) {
```

Comment:
We already have `pointcloud` as a geometry implemented with `BasicOctree`, which implements the collision functions already but since it's a KDTree it's a lot more efficient than checking every point.

https://github.com/viamrobotics/rdk/blob/main/pointcloud/basic_octree.go

Rather than adding `points` as a separate type could this function use something like

```
	pc := pointcloud.NewBasicEmpty()
	for _, point := range points {
		pc.Set(point, pointcloud.NewBasicData())
	}
	return pc.ToBasicOctree(pc, 0)
```
?

Reply:
Thinking more about using `box` I think I might actually walk back my prior suggestion; collision detection is one of our very performance-sensitive sections of code, and using a `box` as a line would be inherently inefficient as we would be checking two non-existent dimensions.

Reply:
The correct thing would be instead, a capsule with radius 0 (something not currently allowed but that's an arbitrary restriction that can be taken off)

Reply:
Instead of using `NewCapsule` I can use the `newCapsuleWithSegPoints` directly to skip validation. That way we can keep the current validation as is.